ekncontent/ekncontent/eknc-content-object-model.c

```diff
@@ -0,0 +1,578 @@
+  /**
+   * EkncContentObjectModel:content-type:
+   *
+   * The source content"s mimetype
+   */
+  eknc_content_object_model_props[PROP_CONTENT_TYPE] =
+    g_param_spec_string ("content-type", "Object Content Type",
+      "Mimetype of the source content",
+      "", G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS);
+  /**
+   * EkncContentObjectModel:synopsis:
+   *
+   * The synopsis for this content object. Defaults to an empty string.
+   */
+  eknc_content_object_model_props[PROP_SYNOPSIS] =
+    g_param_spec_string ("synopsis", "Synopsis",
+      "Synopsis of the document or media object",
+      "", G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS);
+  /**
+   * EkncContentObjectModel:last-modified-date:
+   *
+   * The date of last modification for this content object. It treats dates
+   * according to the ISO8601 standard.
+   */
+  eknc_content_object_model_props[PROP_LAST_MODIFIED_DATE] =
+    g_param_spec_string ("last-modified-date", "Last Modified Date",
+      "Last Modified Date of the document or media object",
+      "", G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS);
+  /**
+   * EkncContentObjectModel:license:
+   *
+   * The license for this content object. Defaults to an empty string.
+   */
+  eknc_content_object_model_props[PROP_LICENSE] =
+    g_param_spec_string ("license", "License",
+      "License of the document or media object",
+      "", G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS);
+  /**
+   * EkncContentObjectModel:featured:
+   *
+   * Whether this content should be given priority in the UI
+   */
+  eknc_content_object_model_props[PROP_FEATURED] =
+    g_param_spec_boolean ("featured", "Featured",
+      "Whether this content should be given priority in the UI",
+      FALSE, G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS);
+  /**
+   * EkncContentObjectModel:tags:
+   *
+   * A list of tag strings associated with the model.
+   */
+  eknc_content_object_model_props[PROP_TAGS] =
+    g_param_spec_variant ("tags", "Tags",
+      "A list of tag strings associated with the model.",
+      G_VARIANT_TYPE ("as"), NULL,
+      G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS);
+  /**
+   * EkncContentObjectModel:resources:
+   *
+   * A list of ekn ids of resources belonging to the model.
+   */
+  eknc_content_object_model_props[PROP_RESOURCES] =
+    g_param_spec_variant ("resources", "Resources",
+      "A list of ekn ids of resources belonging to the model.",
+      G_VARIANT_TYPE ("as"), NULL,
+      G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS);
+
+  g_object_class_install_properties (object_class,
+                                     NPROPS,
+                                     eknc_content_object_model_props);
+}
+
+static void
+eknc_content_object_model_init (EkncContentObjectModel *self)
+{
+}
+
+void
```

Comment:
Well, add a docstring in any case. (I'm surprised gtk-doc isn't marking it as missing.) If you don't `(skip)` then add real documentation as well :wink: otherwise it can be a skeleton docstring.